proxy-node-translator/src/test/java/uk/gov/ida/notification/translator/HubResponseTranslatorTest.java

```diff
@@ -0,0 +1,79 @@
+package uk.gov.ida.notification.translator;
+
+import io.dropwizard.testing.junit.ResourceTestRule;
+import org.junit.ClassRule;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.mockito.junit.MockitoJUnitRunner;
+import uk.gov.ida.common.ExceptionType;
+import uk.gov.ida.exceptions.ApplicationException;
+import uk.gov.ida.notification.contracts.HubResponseTranslatorRequest;
+import uk.gov.ida.notification.contracts.verifyserviceprovider.TranslatedHubResponseBuilder;
+import uk.gov.ida.notification.shared.Urls;
+import uk.gov.ida.notification.shared.proxy.VerifyServiceProviderProxy;
+import uk.gov.ida.notification.translator.resources.HubResponseTranslatorResource;
+import uk.gov.ida.notification.translator.saml.EidasResponseGenerator;
+import uk.gov.ida.saml.core.test.builders.ResponseBuilder;
+
+import javax.ws.rs.client.Entity;
+import javax.ws.rs.core.MediaType;
+import javax.ws.rs.core.Response;
+import java.net.URI;
+import java.util.UUID;
+
+import static org.junit.Assert.assertEquals;
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.when;
+import static uk.gov.ida.saml.core.test.TestCertificateStrings.STUB_COUNTRY_PUBLIC_PRIMARY_CERT;
+
+@RunWith(MockitoJUnitRunner.class)
+public class HubResponseTranslatorTest {
+
+    private final static EidasResponseGenerator eidasResponseGenerator = mock(EidasResponseGenerator.class);
+
+    private final static VerifyServiceProviderProxy verifyServiceProviderProxy = mock(VerifyServiceProviderProxy.class);
+
+    @ClassRule
+    public static final ResourceTestRule resources =
+            ResourceTestRule.builder().addResource(new HubResponseTranslatorResource(eidasResponseGenerator, verifyServiceProviderProxy)).build();
+
+    @Test
+    public void shouldReceive500FromTranslatorWhenEidasResponseNull() {
```

Comment:
I'm a bit confused by this test - it doesn't test what it says it does.
The request passed to the resource has a null `samlResponse` but then the VSP call is stubbed to return a valid response - so that cancels out the null SAML response. The test actually fails because the `eidasResponseGenerator.generate` is not stubbed.

In "real life" the VSP would return an error status if passed a null SAML message - perhaps a 200 + `REQUEST_ERROR` - we should test for all possible responses from the VSP rather than guessing, though

Reply:
These tests show what happens when there are errors in the Translator resource.  The values passed in aren't really important in this case as that's not what's being tested.  It's a valid scenario for testing though so can add in.

Reply:
I was only referring to the `shouldReceive500FromTranslatorWhenEidasResponseNull` test (I had pasted the wrong lines earlier)
It does not test what the test name suggests and in fact it does not test a realistic scenario. At the moment, it's failing because a dependency hasn't been properly mocked - that's not a valid scenario in real usage

Reply:
If you want to test unexpected errors in the Translator that's a valid scenario but make it explicit and indicate that in the test name. So, e.g. 
```
 when(eidasTranslator.generate(...)).thenThrow(....)
```

Reply:
As the eidasResponseGenerator is not mocked it returns null.  It's not something you would want to happen, but it tests what happens if something unexpected goes wrong.  Maybe it needs a different name?

Reply:
Yes, it needs renaming and it would also be better to explicitly simulate a failure/error rather then relying on a side effect of not using a testing framework as it's intended to be used

Reply:
Ok I will do something explicit. I did try adding an explicit 'when' to simulate returning a null, however Mockito then gave an error message saying that the mock was unnecessary, because  the default behaviour is actually to return null.

Reply:
OK that's fine. We can just explicitly throw an exception - this is what we want to test. And rename the test method to indicate what it's validating (handling unexpected/unhandled/internal errors)

Reply:
Removed these tests as Mockito didn't like doing it like this and said the mocks were unnecessary.  Error scenarios are covered by the AppRuleTests which also use the ExceptionMappers.